lib/host-sp-messages/src/lib.rs

```diff
@@ -0,0 +1,221 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Types for messages exchanged between the SP and the host CPU over the
+//! control uart; see RFD 316.
+
+#![cfg_attr(not(test), no_std)]
+
+use hubpack::SerializedSize;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use unwrap_lite::UnwrapLite;
+
+pub use hubpack::error::Error as HubpackError;
+
+/// Maximum message length.
+///
+/// Does not include framing overhead for packetization (e.g., cobs).
+pub const MAX_MESSAGE_SIZE: usize = 1024;
+
+pub mod version {
+    pub const V1: u32 = 1;
+}
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+pub struct Header {
+    pub version: u32,
+    pub sequence: u64,
+    // Followed by either `SpToHost` or `HostToSp`, then any binary data blob
+    // (depending on the variant of those enums), then a 16-bit checksum.
+}
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+pub enum HostToSp {
+    // Microoptimization: insert a dummy variant first, so we never serialize a
+    // command value of `0` to make COBS's life slightly easier.
+    _Unused,
+    GetBootStorageUnit,
+    HostBootFailure {
+        reason: u8,
+    },
+    HostPanic {
+        status: u16,
+        cpu: u16,
+        thread: u64,
+        // Followed by a binary data blob (panic message?)
+    },
+    GetIdentity,
+    GetStatus,
+    /// Instruct SP to perform `status &= mask`
+    ClearStatus {
+        mask: u64,
+    },
+    GetMacAddresses,
+    RebootHost,
+    PowerOffHost,
+    RotRequest, // Followed by a binary data blob (the request)
+}
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+// TODO phase 2 image-related response(s)
+pub enum SpToHost {
+    // Microoptimization: insert a dummy variant first, so we never serialize a
+    // command value of `0` to make COBS's life slightly easier.
+    _Unused,
+    Ack,
+    DecodeFailure(DecodeFailureReason),
+    BootStorageUnit(u8),
+    Identity {
+        model: u8,
+        revision: u8,
+        serial: [u8; 11], // TODO serial format?
+    },
+    Status(u64), // TODO replace u64 with bitflags type?
+    MacAddresses([[u8; 6]; 16]), // TODO send as a data blob? how fixed is 16?
```

Comment:
Vague thoughts:

It might be cleaner to have `GetMacAddress(MacAddressType)` in the `enum HostToSp`, with a separate
```rust
enum MacAddressType {
    SpManagement1,
    SpManagement2,
    HostNic1,
    ...
}
```
(basically forcing the caller to ask for a specific device's MAC address)

Reply:
I'll transport this to a comment on RFD 316 :+1:

Reply:
In the prediscussion RFD316 I have:
> Could also send just first and last address, or start and length.
The reference I used to determine the block of 16 was https://rfd.shared.oxide.computer/rfd/0250#_mac_address_selection
this will drop out when the RFD goes to discussion and we can firm it up.